cmd/main.go

```diff
@@ -1,34 +1,36 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"os"
+	"strings"
 	"weather"
 )
 
-type Response struct {
-	OneWord string  `json:"oneword"`
-	Celcius float64 `json:"celcius"`
-}
-
 func main() {
 
-	location := flag.String("location", "", "a city")
-	flag.Parse()
+	var command strings.Builder
 
-	if len(*location) == 0 {
-		fmt.Printf("please enter a location\n")
+	if len(os.Args) == 1 {
+		fmt.Fprintf(os.Stderr, "please set a location e.g. london\n")
 		os.Exit(2)
 	}
-	flag.Parse()
+
+	for i, v := range os.Args {
```

Comment:
This doesn't feel like something we should be doing in `main`, does it? We'd like to have some function like `LocationFromArgs`, or something like that, which takes a `[]string`, and we could pass `os.Args` to it. That way, it's easy to test with various different inputs.

Reply:
fixed